cli/command/stack/deploy_composefile.go

```diff
@@ -320,8 +320,20 @@ func deployServices(
 
 			if resolveImage == resolveImageAlways || (resolveImage == resolveImageChanged && image != service.Spec.Labels[convert.LabelImage]) {
 				updateOpts.QueryRegistry = true
+			} else {
```

Comment:
Argh, you're right; changed it to the if/else based on your previous comment. I'm not entirely convinced using a "switch" is much clearer; wether or not to keep the existing information is based on `updateOpts.QueryRegistry ` being `true`, which was why I had it like that in the first implementation